packages/components/src/components/settings.tsx

```diff
@@ -0,0 +1,142 @@
+import { PlusIcon, XIcon } from '@heroicons/react/solid';
+import { useAtom } from 'jotai';
+import { SidebarOverlay } from './sidebarOverlay';
+import { headersAtom } from './tab/store';
+import { Headers as HeadersType } from "./tab/types"
+
+interface SettingsProps {
+  hide: () => void
+}
+export const Settings: React.FC<SettingsProps> = ({ hide }) => {
+  return (
+    <SidebarOverlay hide={hide}>
+      <div className='bg-primary p-3 sticky top-0 shadow-md flex justify-between items-center'>
+        <h1 className='text-2xl font-semibold'>
+          Settings
+        </h1>
+
+        <button onClick={hide}>
+          <XIcon
+            width={20}
+            height={20}
+            className="text-neutral-300 hover:text-white transition-colors"
+          />
+        </button>
+      </div>
+      <div className="p-3 py-4">
+        <h2 className="text-lg font-semibold">
+          Global headers
+        </h2>
+        <Headers />
+      </div>
+    </SidebarOverlay>
+  );
+}
+
+const Headers = () => {
+  const [headers, setHeaders] = useAtom(headersAtom)
+  let headerItems = Object.entries(headers)
+
+  return (
+    <>
+      {headerItems.map(([name, value], idx) => (
+        <Header
+          removeHeader={() => {
+            const newHeaders = headerItems
+              .filter((_item, headerIdx) => idx !== headerIdx)
+              .reduce((prev, curr) => {
+                prev[curr[0]] = curr[1]
+                return prev
+              }, {} as HeadersType)
+
+            setHeaders(() => newHeaders)
```

Comment:
use the `headers` passed as an argument to `setHeaders` instead of the atom to ensure consistency (this applies to all the methods here). Also don't compare indexes as those are unstable. Instead, compare something unique. Also use `splice` to remove something from an array.

Reply:
removed